### PR TITLE
[Spark] Support clone and restore for Identity Columns

### PR DESCRIPTION
**Pull request by zhipengmao-db**
#### Which Delta project/connector is this regarding?

- [x] Spark
- [ ] Standalone
- [ ] Flink
- [ ] Kernel
- [ ] Other (fill in here)

## Description
This PR is part of https://github.com/delta-io/delta/issues/1959 .
It adds support for clone and restore tables with identity columns.

## How was this patch tested?

Clone and restore related test cases.

## Does this PR introduce _any_ user-facing changes?

Users will be able to call clone and restore on tables with identity columns.

### CODE REVIEW

**Review by larsk-db — APPROVED** [review]
LGTM

File: spark/src/main/scala/org/apache/spark/sql/delta/commands/CloneTableBase.scala [hunk 1]
```diff
@@ -314,7 +314,12 @@ abstract class CloneTableBase(
     clonedMetadata = clonedMetadata.copy(configuration =
       clonedMetadata.configuration.filterKeys(
         !CoordinatedCommitsUtils.TABLE_PROPERTY_KEYS.contains(_)).toMap)
-    clonedMetadata
+
+    val clonedSchema =
+      IdentityColumn.copySchemaWithMergedHighWaterMarks(
+        schemaToCopy = clonedMetadata.schema,
+        schemaWithHighWaterMarksToMerge = targetSnapshot.metadata.schema)
+    clonedMetadata.copy(schemaString = clonedSchema.json)
```
Comment (c27kwan) [inline spark/src/main/scala/org/apache/spark/sql/delta/commands/CloneTableBase.scala:322]: We can combine the two `cloneMetadata.copy()` calls. We should make `cloneConfigurations` a separate variable for improved readability.

**Comment by zhipengmao-db** [comment]
The error in spark master test is not related to the change as it fails in other PRs as well.

```
[error] /home/runner/work/delta/delta/spark-connect/server/src/test/scala-spark-master/io/delta/connect/DeltaConnectPlannerSuite.scala:60:58: value forTesting is not a member of object org.apache.spark.sql.connect.service.SessionHolder
[error]       val result = new SparkConnectPlanner(SessionHolder.forTesting(spark)).transformRelation(input)
[error]                                                          ^
[error] /home/runner/work/delta/delta/spark-connect/server/src/test/scala-spark-master/io/delta/connect/DeltaConnectPlannerSuite.scala:86:58: value forTesting is not a member of object org.apache.spark.sql.connect.service.SessionHolder
[error]       val result = new SparkConnectPlanner(SessionHolder.forTesting(spark)).transformRelation(input)
[error]                                                          ^
[warn] /home/runner/work/delta/delta/spark-connect/server/src/test/scala-spark-master/io/delta/connect/DeltaConnectPlannerSuite.scala:27:68: Unused import
[warn] import org.apache.spark.sql.connect.delta.ImplicitProtoConversions._
[warn]                   
```

File: spark/src/main/scala/org/apache/spark/sql/delta/commands/CloneTableBase.scala [hunk 2]
```diff
@@ -311,10 +311,14 @@ abstract class CloneTableBase(
     }
 
     // Coordinated Commit configurations are never copied over to the target table.
-    clonedMetadata = clonedMetadata.copy(configuration =
-      clonedMetadata.configuration.filterKeys(
-        !CoordinatedCommitsUtils.TABLE_PROPERTY_KEYS.contains(_)).toMap)
-    clonedMetadata
+    val filteredConfigeration = clonedMetadata.configuration
```
Comment (c27kwan) [inline spark/src/main/scala/org/apache/spark/sql/delta/commands/CloneTableBase.scala:314]:
```suggestion
    val filteredConfiguration = clonedMetadata.configuration
```

**Review by c27kwan — APPROVED** [review]
lgtm! Good work :)

**Comment by vkorukanti** [comment]
Spark Master failures are already known issue.

**Pull request merged**